hoomd/mpcd/CollisionMethod.cc

```diff
@@ -395,13 +419,94 @@ void mpcd::CollisionMethod::transferRigidBodyMomenta(uint64_t timestep)
             {
             angmom_change_body.z = Scalar(0);
             }
+
+        // calculate scaling factor
+        Scalar a = 0;
+        Scalar b = 0;
+        Scalar c = 0;
+        Scalar s = 0;
+
+        const quat<Scalar> initial_angmom_body = Scalar(0.5) * conj(orientation) * angmom;
+        if (inertia.x > Scalar(0))
+            {
+            a += angmom_change_body.x * angmom_change_body.x / inertia.x;
+            b += initial_angmom_body.v.x * angmom_change_body.x / inertia.x;
+            }
+
+        if (inertia.y > Scalar(0))
+            {
+            a += angmom_change_body.y * angmom_change_body.y / inertia.y;
+            b += initial_angmom_body.v.y * angmom_change_body.y / inertia.y;
+            }
+
+        if (inertia.z > Scalar(0))
+            {
+            a += angmom_change_body.z * angmom_change_body.z / inertia.z;
+            b += initial_angmom_body.v.z * angmom_change_body.z / inertia.z;
+            }
+        a *= Scalar(0.5);
+        c = ke_tra_change - ke_accum;
+
+        // check if there are imaginary roots
+        Scalar d = b * b - 4 * a * c;
+        if (d < 0.0)
+            {
+            errors.x = 1;
+            m_errors.resetFlags(errors);
```

Comment:
This can come after the for loop, not inside it

Reply:
How? If d is negative, then the for loop needs to be exited so that the errors can be thrown. Or do you mean it should break instead and put `m_errors.resetFlags(errors);` after the loop?